digitalocean/partnerinterconnectattachment/resource_partner_interconnect_attachment.go

```diff
@@ -0,0 +1,289 @@
+package partnerinterconnectattachment
+
+import (
+	"context"
+	"fmt"
+	"log"
+	"net/http"
+	"time"
+
+	"github.com/digitalocean/godo"
+	"github.com/digitalocean/terraform-provider-digitalocean/digitalocean/config"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/diag"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/retry"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/validation"
+)
+
+func ResourceDigitalOceanPartnerInterconnectAttachment() *schema.Resource {
+	return &schema.Resource{
+		CreateContext: resourceDigitalOceanPartnerInterconnectAttachmentCreate,
+		ReadContext:   resourceDigitalOceanPartnerInterconnectAttachmentRead,
+		UpdateContext: resourceDigitalOceanPartnerInterconnectAttachmentUpdate,
+		DeleteContext: resourceDigitalOceanPartnerInterconnectAttachmentDelete,
+		Importer: &schema.ResourceImporter{
+			StateContext: schema.ImportStatePassthroughContext,
+		},
+
+		Schema: map[string]*schema.Schema{
+			"id": {
+				Type:        schema.TypeString,
+				Computed:    true,
+				Description: "The ID of the Partner Interconnect Attachment",
+			},
+			"name": {
+				Type:         schema.TypeString,
+				Required:     true,
+				Description:  "The name of the Partner Interconnect Attachment",
+				ValidateFunc: validation.NoZeroValues,
+			},
+			"connection_bandwidth_in_mbps": {
+				Type:         schema.TypeInt,
+				Required:     true,
+				Description:  "The connection bandwidth in Mbps",
+				ValidateFunc: validation.NoZeroValues,
+			},
+			"region": {
+				Type:         schema.TypeString,
+				Required:     true,
```

Comment:
```suggestion
				Required:     true,
				ForceNew: true,
```

Generally, if an attribute is not updateable, it should be set to `ForceNew: true`. Probably want this on `connection_bandwidth_in_mbps` and `naas_provider` too.